src/analyzer-range-workspace/analyzer-range.repository.ts

```diff
@@ -48,6 +48,19 @@ export class AnalyzerRangeWorkspaceRepository extends Repository<
     return null;
   }
 
+  async getAnalyzerRangeByLogicalKey(
+    componentId: string,
+    beginDate: Date,
+    beginHour: number,
+  ): Promise<AnalyzerRange | null> {
+    return this.createQueryBuilder('ar')
+      .innerJoin('ar.component', 'c')
+      .where('c.id = :componentId', { componentId })
```

Comment:
I don't think this join is necessary, the component fields aren't being selected and the component ID is already stored in the `analyzer_range` table.
```suggestion
      .where('ar.componentRecordId = :componentId', { componentId })
```